packages/e2e-test-utils/src/inserter.js

```diff
@@ -100,157 +142,208 @@ async function waitForInserterCloseAndContentFocus() {
 /**
- * Wait for the inserter search to yield results because that input is debounced.
+ * Searches for an entity matching given category and term via the global
+ * inserter. If nothing is found, null will be returned.
+ *
+ * Available categories: Blocks, Patterns, Reusable and Block Directory.
+ *
+ * @param {string} category   The category to search within.
+ * @param {string} searchTerm The text to search the inserter for.
+ * @return {Promise<ElementHandle|null>} The handle of the element to be inserted or
+ * null if nothing was found.
  */
-async function waitForInserterSearch() {
+export async function searchGlobalInserter( category, searchTerm ) {
+	await openGlobalBlockInserter();
+	await page.waitForSelector( INSERTER_SEARCH_SELECTOR );
+	await page.focus( INSERTER_SEARCH_SELECTOR );
+	await pressKeyWithModifier( 'primary', 'a' );
+	await page.keyboard.type( searchTerm );
+
+	// Wait for the default block list to disappear to prevent its items from
+	// being considered as search results. This is needed since we're debouncing
+	// search request.
+	await page.waitForSelector( '.block-editor-inserter__block-list', {
+		hidden: true,
+	} );
+
+	let waitForInsertElement;
+	let waitForNoResults;
+
+	switch ( category ) {
+		case 'Blocks':
+		case 'Patterns':
+		case 'Reusable': {
+			waitForInsertElement = async () => {
+				return await page.waitForXPath(
+					`//*[@role='option' and contains(., '${ searchTerm }')]`
+				);
+			};
+			waitForNoResults = async () => {
+				await page.waitForSelector(
+					'.block-editor-inserter__no-results'
+				);
+				return null;
+			};
+			break;
+		}
+		case 'Block Directory': {
+			waitForInsertElement = async () => {
+				// Return the first item from the Block Directory search results.
+				return await page.waitForSelector(
+					'.block-directory-downloadable-blocks-list button:first-child'
+				);
+			};
+			waitForNoResults = async () => {
+				// Use a soft timeout if Block Directory doesn't return anything
+				// within 5 seconds, as there's no "empty results" element being
+				// rendered when nothing is found.
+				return await new Promise( ( resolve ) =>
+					setTimeout( () => resolve( null ), 5000 )
+				);
+			};
+		}
+	}
+
+	return await Promise.race( [ waitForInsertElement(), waitForNoResults() ] );
+}
+
+/**
+ * Inserts an entity matching given category and term via the global inserter.
+ * If the entity is not instantly available in the open inserter, a search will
+ * be performed. If the search returns no results, an error will be thrown.
+ *
+ * Available categories: Blocks, Patterns, Reusable and Block Directory.
+ *
+ * @param {string} category   The category to insert from.
+ * @param {string} searchTerm The text to search the inserter for.
+ */
+export async function insertFromGlobalInserter( category, searchTerm ) {
+	await openGlobalBlockInserter();
+	await selectGlobalInserterTab( category );
+
+	let insertButton;
+
+	// Check if the block is instantly insertable...
```

Comment:
We seem to be referring it to as "entity" in many areas, which maybe makes more sense since this is also responsible for inserting block patterns.

Reply:
Ah, I actually meant "blocks" because only items from the Blocks and Reusable (Blocks) are instantly insertable after switching to their tabs. I should have made this clearer though, so I think wrapping this block with an `if ( block OR reusable )` would do the trick. Nice catch!

Reply:
Updated in https://github.com/WordPress/gutenberg/pull/46366/commits/c0d7109adcbd5ed5f901e9006990522f3ca3028e